boxsdk/object/base_api_json_object.py

```diff
@@ -9,22 +9,68 @@
 
 class BaseAPIJSONObjectMeta(type):
     """
-    Metaclass for Box API objects. Registers classes so that API responses can be translated to the correct type.
-    Relies on the _item_type field defined on the classes to match the type property of the response json.
-    But the type-class mapping will only be registered if the module of the class is imported.
-    So it's also important to add the module name to __all__ in object/__init__.py.
+    Metaclass for Box API objects.
+
+    Registers classes with the default translator, so that API responses can be
+    translated to the correct type. This relies on the _item_type field, which
+    must be defined in the class's namespace dict (and must be re-defined, in
+    order to register a custom subclass), to match the 'type' field of the
+    response json.  But the type-class mapping will only be registered if the
+    module of the class is imported.
+
+    For example, events returned from the API look like
+
+    .. code-block:: json
+
+        {'type': 'event', ...}
+
+    so a class for that type could be created and registered with the default
+    translator like this:
+
+    .. code-block:: python
+
+        class Event(BaseAPIJSONObject):
+            _item_type = 'event'
+            ...
+
+    NOTE: The default translator registration functionality is a private
+    implementation detail of the SDK, to make it easy to register the default
+    API object classes with the default translator. For convenience and
+    backwards-compatability, developers are allowed to re-define the _item_type
+    field in their own custom subclasses in order to take advantage of this
+    functionality, but are encouraged not to. Since this is a private
+    implementation detail, it may change or be removed in any major or minor
+    release. Additionally, it has the usual hazards of mutable global state.
+    The supported and recommended ways for registering custom subclasses are:
+
+    - Constructing a new :class:`Translator`, calling `Translator.register()`
+      as necessary, and passing it to the :class:`BoxSession` constructor.
+    - Calling `session.translator.register()` on an existing
+      :class:`BoxSession`.
+    - Calling `client.translator.register()` on an existing :class:`Client`.
     """
+
     def __init__(cls, name, bases, attrs):
         super(BaseAPIJSONObjectMeta, cls).__init__(name, bases, attrs)
-        item_type = getattr(cls, '_item_type', None)
+        item_type = attrs.get('_item_type', None)
         if item_type is not None:
-            Translator().register(item_type, cls)
+            Translator._default_translator.register(item_type, cls)   # pylint:disable=protected-access
```

Comment:
When I started this work, I started it with `Translator.default_translator`. At some point I decided to switch it to `Translator._default_translator`, to discourage applications from using it.

I could add `Translator.register_default`, but then that might encourage people to use it.

If we're going to make something public, I think it'd be more useful to make `Translator.default_translator` public, rather than making it private but adding this public method. Unless you think there's utility in only making the `register` method public.

I think it's okay for us to access the protected attribute, since it's clearly documented as being for internal use.